webapp/channels/src/components/user_settings_modal/advanced_display_settings/advanced_display_settings.tsx

```diff
@@ -0,0 +1,262 @@
+    [AdvanceSettings.CODE_BLOCK_CTRL_ENTER]: string;
+    [AdvanceSettings.FORMATTING]: string;
+    [AdvanceSettings.JOIN_LEAVE]: string;
+    [AdvanceSettings.NAME_DISABLE_TYPING_MESSAGES]: string;
+    [AdvanceSettings.NAME_DISABLE_TELEMETRY]: string;
+    [AdvanceSettings.NAME_DISABLE_CLIENT_PLUGINS]: string;
+    [AdvanceSettings.UNREAD_SCROLL_POSITION]: string;
+    [AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED]: string;
+}
+
+export type OwnProps = {
+    userId?: UserProfile['id'];
+}
+
+export type Props = OwnProps & PropsFromRedux;
+
+export default function AdvancedDisplaySettings(props: Props) {
+    const [haveChanges, setHaveChanges] = useState(false);
+    const {formatMessage} = useIntl();
+
+    const [settings, setSettings] = useState<SettingsType>({
+        [AdvanceSettings.SEND_ON_CTRL_ENTER]: props.sendOnCtrlEnter,
+        [AdvanceSettings.CODE_BLOCK_CTRL_ENTER]: props.codeBlockOnCtrlEnter,
+        [AdvanceSettings.FORMATTING]: props.formatting,
+        [AdvanceSettings.JOIN_LEAVE]: props.joinLeave,
+        [AdvanceSettings.NAME_DISABLE_TYPING_MESSAGES]: props.disableTypingMessages,
+        [AdvanceSettings.NAME_DISABLE_TELEMETRY]: props.disableTelemetry,
+        [AdvanceSettings.NAME_DISABLE_CLIENT_PLUGINS]: props.disableClientPlugins,
+        [AdvanceSettings.UNREAD_SCROLL_POSITION]: props.unreadScrollPosition,
+        [AdvanceSettings.SYNCED_DRAFTS_ARE_ALLOWED]: props.syncedDraftsAreAllowed ? 'true' : 'false',
+    });
+
+    const handleChange = useCallback((values: Record<string, string>) => {
+        setSettings({...settings, ...values});
+        setHaveChanges(true);
+    }, [settings]);
+
+    const handleSubmit = async (): Promise<void> => {
+        const preferences: PreferenceType[] = [];
+        const {actions, userId} = props;
+
+        Object.keys(settings).forEach((setting) => {
+            let category = Constants.Preferences.CATEGORY_ADVANCED_SETTINGS;
+            if (setting === Preferences.NAME_DISABLE_CLIENT_PLUGINS ||
+                setting === Preferences.NAME_DISABLE_TELEMETRY ||
+                setting === Preferences.NAME_DISABLE_TYPING_MESSAGES
+            ) {
+                category = Preferences.CATEGORY_PERFORMANCE_DEBUGGING;
+            }
+            preferences.push({
+                user_id: userId,
+                category,
+                name: setting,
+                value: settings[setting],
+            });
+        });
+
+        await actions.savePreferences(userId, preferences);
+        setHaveChanges(false);
+    };
+
+    const ctrlSendTitleAndDesc = getCtrlSendText();
+    const ctrlSendContent = (
+        <RadioSettingItem
+            inputFieldValue={getCtrlSelectedOption(settings[AdvanceSettings.SEND_ON_CTRL_ENTER], settings[AdvanceSettings.CODE_BLOCK_CTRL_ENTER])}
+            inputFieldData={ctrlSendInputFieldData}
+            handleChange={(e) => handleChange(getCtrlSendPreferenceValue(e.target.value))}
+        />
+    );
```

Comment:
I'm not a huge fan of having the definition of each setting split between this file and a `advanced_display_settings/utils.tsx` which makes reading it hard and it means we're passing a lot of anonymous functions for each setting's `handleChange`.

Instead of doing that, what if we instead make a component for each of these settings sections? That way, the component could encapsulate all of the logic for setting its state and rendering itself while keeping `AdvancedDisplaySettings` relatively by not having it care about that logic. I'm thinking something like
```ts
// advanced_display_settings/ctrl_send_section.tsx

// This file handles everything involved in the controls for this setting, the text shown to the user, and
// for showing/modifying the values stored in the parent component's state. Helpers specific to this setting
// like getCtrlSendText and getCtrlSendPreferenceValue could be moved into this file instead of having them
// all mixed together in advanced_display_settings/utils.tsx

export default function CtrlSendSection(props: Props) {
    const {
        sendOnCtrlEnter,
        codeBlockCtrlEnter,
        onChange,
    } = props;

    const ctrlSendTitleAndDesc = getCtrlSendText();

    const handleChange = useCallback((e) => {
        onChange(getCtrlSendPreferenceValue(e.target.value));
    }, [onChange]);

    return (
        <SectionCreator
            title={formatMessage(ctrlSendTitleAndDesc.ctrlSendTitle)}
            content={
                <RadioSettingItem
                    inputFieldValue={getCtrlSelectedOption(sendOnCtrlEnter, codeBlockCtrlEnter)}
                    inputFieldData={ctrlSendInputFieldData}
                    handleChange={handleChange}
                />
            }
            description={formatMessage(ctrlSendTitleAndDesc.ctrlSendDesc)}
        />
    };
}

// advanced_display_settings/advanced_display_settings.tsx

// Now, this only needs to import CtrlSendSection without all of the extra helpers

export default function AdvancedDisplaySettings(props: Props) {
    const [settings, setSettings] = useState<SettingsType>({...});

    const handleSubmit = useCallback(() => {...}, [userId, savePreferences]);

    return (
        <>
            <CtrlSendSection
                sendOnCtrlEnter={settings[AdvanceSettings.SEND_ON_CTRL_ENTER]
                codeBlocKCtrlEnter={settings[AdvanceSettings.CODE_BLOCK_CTRL_ENTER]}
                onSubmit={handleSubmit}
            />
        </>
    );
}
```

Generally, I want to try to organize this code together by it's purpose (eg. "these functions are for the send with ctrl + enter setting") rather than by the type of code (eg. "these are utility functions for all of the advanced settings") like you have them currently.

Hopefully that makes sense since I'm worried I'm not explaining it great 😅